tests/services/order_validation_service/test_case_sample_rules.py

```diff
@@ -62,14 +131,212 @@ def test_missing_required_volume(valid_order: OrderWithCases):
+    existing_sample = ExistingSample(internal_id="Non-existent sample", status=StatusEnum.unknown)
+    valid_order.cases[0].samples.append(existing_sample)
+
+    # WHEN validating that the samples exists
+    errors = validate_samples_exist(order=valid_order, store=store_with_multiple_cases_and_samples)
+
+    # THEN an error should be returned
+    assert errors
+
+    # THEN the error should concern the non-existent sample
+    assert isinstance(errors[0], SampleDoesNotExistError)
+
+
+def test_application_is_incompatible(
+    valid_order: TomteOrder, sample_with_non_compatible_application: TomteSample, base_store: Store
+):
+
+    # GIVEN an order that has a sample with an application which is incompatible with the workflow
+    valid_order.cases[0].samples.append(sample_with_non_compatible_application)
+
+    # WHEN validating the order
+    errors = validate_application_compatibility(order=valid_order, store=base_store)
+
+    # THEN an error should be returned
+    assert errors
+
+    # THEN the error should be about the application compatability
+    assert isinstance(errors[0], ApplicationNotCompatibleError)
+
+
+def test_subject_ids_same_as_case_names_not_allowed(valid_order: TomteOrder):
+
+    # GIVEN an order with a sample having its subject_id same as the case's name
+    case_name = valid_order.cases[0].name
+    valid_order.cases[0].samples[0].subject_id = case_name
+
+    # WHEN validating that no subject ids are the same as the case name
+    errors = validate_subject_ids_different_from_case_names(valid_order)
+
+    # THEN an error should be returned
+    assert errors
+
+    # THEN the error should be concerning the subject id being the same as the case name
+    assert isinstance(errors[0], SubjectIdSameAsCaseNameError)
+
+
+def test_well_position_missing(
+    valid_order: TomteOrder, sample_with_missing_well_position: TomteSample
+):
+    # GIVEN an order with a sample with a missing well position
+    valid_order.cases[0].samples.append(sample_with_missing_well_position)
+
+    # WHEN validating that no well positions are missing
+    errors = validate_well_positions_required(valid_order)
+
+    # THEN an error should be returned
+    assert errors
+
+    # THEN the error should concern the missing well position
+    assert isinstance(errors[0], WellPositionMissingError)
+
+
+def test_container_name_missing(
+    valid_order: TomteOrder, sample_with_missing_container_name: TomteSample
+):
+
+    # GIVEN an order with a sample missing its container name
+    valid_order.cases[0].samples.append(sample_with_missing_container_name)
+
+    # WHEN validating that it is not missing any container names
+    errors = validate_container_name_required(order=valid_order)
+
+    # THEN an error should be raised
+    assert errors
+
+    # THEN the error should concern the missing container name
+    assert isinstance(errors[0], ContainerNameMissingError)
+
+
+@pytest.mark.parametrize("sample_volume", [1, 200])
```

Comment:
It would be useful to add ids with the meaning of these values, I guess one is too low and one is too high